torch/csrc/cuda/Module.cpp

```diff
@@ -1388,6 +1388,24 @@ static void registerCudaPluggableAllocator(PyObject* module) {
             device, mempool_id, [](cudaStream_t) { return true; });
       });
 
+  m.def(
+      "_cuda_beginAllocateCurrentThreadToPool",
+      [](c10::DeviceIndex device, at::cuda::MempoolId_t mempool_id) {
+        auto tid = std::this_thread::get_id();
+
+        c10::cuda::CUDACachingAllocator::beginAllocateToPool(
+            device, mempool_id, [=](cudaStream_t) {
+              auto current_tid = std::this_thread::get_id();
+              return current_tid == tid;
+            });
+      });
+
+  m.def(
+      "_cuda_endAllocateToPool",
+      [](c10::DeviceIndex device, at::cuda::MempoolId_t mempool_id) {
+        c10::cuda::CUDACachingAllocator::endAllocateToPool(device, mempool_id);
+      });
+
   m.def(
```

Comment:
can you delete `_cuda_endAllocateCurrentStreamToPool` and replace uses with `_cuda_endAllocateToPool`? They have exact same implementation.